### ISSUE TITLE
plpgsql: DETAIL: stmt_dyn_exec is not yet supported

### ISSUE THREAD

**Issue by paigeadelethompson** [cockroachdb/cockroach#137561]
```
CREATE OR REPLACE FUNCTION diesel_manage_updated_at (_tbl regclass)
    RETURNS VOID
    AS $$
BEGIN
    EXECUTE format('CREATE TRIGGER set_updated_at BEFORE UPDATE ON %s
                    FOR EACH ROW EXECUTE PROCEDURE diesel_set_updated_at()', _tbl);
END;
$$
LANGUAGE plpgsql;

CREATE OR REPLACE FUNCTION diesel_set_updated_at ()
    RETURNS TRIGGER
    AS $$
BEGIN
    IF (NEW IS DISTINCT FROM OLD AND NEW.updated_at IS NOT DISTINCT FROM OLD.updated_at) THEN
        NEW.updated_at := CURRENT_TIMESTAMP;
    END IF;
    RETURN NEW;
END;
$$
LANGUAGE plpgsql;
```

> ERROR: unimplemented: attempted to use a PL/pgSQL statement that is not yet supported
> SQLSTATE: 0A000
> DETAIL: stmt_dyn_exec is not yet supported
> HINT: You have attempted to use a feature that is not yet implemented.
> 
> Please check the public issue tracker to check whether this problem is
> already tracked. If you cannot find it there, please report the error
> with details by creating a new issue.
> 
> If you would rather not post publicly, please contact us directly
> using the support form.
> 
> We appreciate your feedback.
> CREATE FUNCTION
> 
> Time: 741ms total (execution 740ms / network 1ms)
> 

Jira issue: CRDB-45652

**Comment by paigeadelethompson** [comment]
> Hi @paigeadelethompson, please add branch-* labels to identify which branch(es) this C-bug affects.
> 
> 🦉 Hoot! I am a [Blathers](https://github.com/apps/blathers-crl), a bot for [CockroachDB](https://github.com/cockroachdb). My owner is [dev-inf](https://github.com/orgs/cockroachdb/teams/dev-inf).

```
Build Tag:        v24.3.0
Build Time:       2024/11/21 17:04:13
Distribution:     CCL
Platform:         linux amd64 (x86_64-pc-linux-gnu)
Go Version:       go1.22.8 X:nocoverageredesign
C Compiler:       gcc 6.5.0
Build Commit ID:  ef2ebe96f3a26fd6d6716e2ec4a0c8c6fe0fb567
Build Type:       release
Enabled Assertions: false
```

**Comment by paigeadelethompson** [comment]
also found the regression test for the `stmt_dyn_exec` but I'm not really that much of an expert in sprocs 

https://github.com/cockroachdb/cockroach/blob/effc6b28a19c516cb378fdba95078ce4d8b5b550/pkg/ccl/logictestccl/testdata/logic_test/plpgsql_unsupported#L27

this is just the first of 242 migrations that need to be applied for this application to work correctly. I don't even know sprocs that well to say whether or not this is something I can workaround (I wanna try but I'm also looking at this list of 241 migrations that need to be applied and thinking my god what am I getting myself into? Moreover I'm not sure how I arrived at the conclusion that sprocs would be limited in scope (My only frame of reference is MSSQL and admittedly it's been a really long time but the only time I had problems like this was going into a migration from one version to another. Is that the point I'm missing here? I know there are different versions of pgsql so I'm wondering if there's a compatibility level for which I should assume cockroach should work fine for or is it always just hit or miss? 

Thanks
-P

**Comment by paigeadelethompson** [comment]
Stale issue, but apparently a more broad effort to address the problem (mine also originates from migrations created with diesel) https://github.com/cockroachdb/cockroach/issues/13787

**Comment by yuzefovich** [comment]
Thanks for opening up the issue! Indeed, you ran into a known limitation that we don't yet support `EXECUTE` commands in PLpgSQL - this work is tracked by #115300. More generally, #117744 is the main tracking issue to see which PLpgSQL features are unimplemented still.

We did implement `RETURNS TABLE` syntax very recently ([last week](https://github.com/cockroachdb/cockroach/pull/137251)), so that feature should be available in 25.1 release coming out early next year.

**Comment by paigeadelethompson** [comment]
Oh cool thank you for clearing that up some, I started looking at the code and found this: https://github.com/cockroachdb/cockroach/commit/75055836952d915a0ba570d49c92020643369505#diff-f8265107fda3aef59cddcab4cdf6b229312224027fe3db5264879cd2659afbbbR164

which looks like it might already be in place (I see a tag that says this is already in `24.3.1`, and it looks like I am running `build:               CCL v24.3.0 @ 2024/11/21 17:04:13 (go1.22.8 X:nocoverageredesign)` which is surprising because I'm running the Docker container `cockroachdb/cockroach   latest    296e6e763a75` 

I'll keep an eye on this maybe this will work with `24.3.1` if that's an option (I'll have to see what I can get from https://hub.docker.com/r/cockroachdb/cockroach/tags?name=latest and it looks like these were updated in the past two days so I'll have to check later.)

**Comment by yuzefovich** [comment]
The commit that you linked only added the parser support, and it merged last year, so it's been included in versions 23.2+ or so. v24.3.1 won't have any new features supported I think - we generally only include new features in major releases (so either 24.3.0 or 25.1.0, but not in the minor release like 24.3.1).

I'm going to close this issue as a dup of #117744 - please follow that issue for updates about PLpgSQL functionality that we add full support for.
